### PR TITLE
Avoid compiling any OpenCl on macos.

### PR DESCRIPTION
**Pull request by porcuquine**
The fixes in #54 were on the 1.2.x branch. This PR applies them to the 2.x branch.

### CODE REVIEW

File: src/batch_hasher.rs [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-#[cfg(all(feature = "gpu", not(target_os = "macos")))]
-use crate::cl;
-
 #[derive(Clone, Copy, Debug)]
 pub enum BatcherType {
-    CustomGPU(GPUSelector),
+    #[cfg(all(feature = "gpu", not(target_os = "macos")))]
+    CustomGPU(cl::GPUSelector),
+    #[cfg(all(feature = "gpu", target_os = "macos"))]
+    CustomGPU(()),
```
Comment (dignifiedquire) [inline src/batch_hasher.rs:15]: this could be just CustomGPU without anything in it

File: src/lib.rs [hunk 2]
```diff
@@ -37,11 +37,8 @@ pub mod column_tree_builder;
 #[cfg(feature = "gpu")]
 mod gpu;
 
-#[cfg(feature = "gpu")]
-pub use gpu::GPUSelector;
-
 #[cfg(all(feature = "gpu", not(target_os = "macos")))]
-mod cl;
+pub mod cl;
```
Comment (dignifiedquire) [inline src/lib.rs:41]: why is this pub now?

File: src/batch_hasher.rs [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
-#[cfg(all(feature = "gpu", not(target_os = "macos")))]
-use crate::cl;
-
 #[derive(Clone, Copy, Debug)]
 pub enum BatcherType {
-    CustomGPU(GPUSelector),
+    #[cfg(all(feature = "gpu", not(target_os = "macos")))]
+    CustomGPU(cl::GPUSelector),
+    #[cfg(all(feature = "gpu", target_os = "macos"))]
+    CustomGPU(),
```
Comment (dignifiedquire) [inline src/batch_hasher.rs:15]: not even those parens are needed

File: src/batch_hasher.rs (hunk 1 above)
Comment (porcuquine) [reply src/batch_hasher.rs]: changed

File: src/lib.rs (hunk 2 above)
Comment (porcuquine) [reply src/lib.rs]: Merge conflict confusion. Changed back.

File: src/batch_hasher.rs (hunk 3 above)
Comment (porcuquine) [reply src/batch_hasher.rs:15]: lol, you are stalking my commits.

Comment (porcuquine) [reply src/batch_hasher.rs:15]: They are needed, though, and so was the content. `CustomCPU()` actually fails on macos (just checked). It still needs to match `CustomCPU(_)` patterns.

Comment (dignifiedquire) [reply src/batch_hasher.rs]: I see, I guess you could work around that, but this is fine

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**